### PR TITLE
Fail setup on no CUDA environment

### PR DESCRIPTION
**Pull request by okuta**

### CODE REVIEW

File: cupy_setup_build.py [hunk 1]
```diff
@@ -38,8 +38,8 @@
             'cupy.util',
         ],
         'include': [
-            'cublas_v2.h',
             'cuda.h',
+            'cublas_v2.h',
```
Comment (unnonouno) [inline cupy_setup_build.py]: Why this change required?

File: cupy_setup_build.py [hunk 2]
```diff
@@ -221,27 +221,29 @@ def make_extensions(options, compiler, use_cython):
...
+                err = True
+            elif('check_method' in module and
+                 not module['check_method'](compiler, settings)):
+                err = True
 
-            if 'check_method' in module and \
-               not module['check_method'](compiler, settings):
-                continue
+            if module['name'] == 'cuda' and err:
+                raise Exception('Please check above error log.')
```
Comment (unnonouno) [inline cupy_setup_build.py]: What happens when cudnn is not found?

**Comment by beam2d** [comment]
After this PR is merged, Travis only checks if CuPy is built without errors (sdist) and passes style checking.

**Comment by beam2d** [comment]
Note: it looks better to remove appveyor from CuPy because there is no CI test that makes differences between environments.
